src/main/java/Duke.java

```diff
@@ -1,10 +1,201 @@
+                lst.add(new ToDo(description, status == 1));
+            } else {
+                String additionalInfo = arr[3].trim();
+                if (type.equals("D")) {
+                    lst.add(new Deadline(description, additionalInfo, status == 1));
+                } else {
+                    lst.add(new Event(description, additionalInfo, status == 1));
+                }
+            }
+        }
+    }
+    public static String readFileContents(String filePath) throws FileNotFoundException {
+        File f = new File(filePath);
+        String content = "";
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            content += s.nextLine() + System.lineSeparator();
+        }
+        return content;
+    }
+    public static Task createTask(String firstWord, String input) throws DukeException {
+        Task newTask;
+        if (firstWord.equals("todo")) {
+            String[] arr = input.split("todo ");
+            if (arr.length == 1) {
+                throw new DukeException("The description of a todo cannot be empty.");
+            } else {
+                String tsk = arr[1].trim();
+                newTask = new ToDo(tsk);
+            }
+        } else if (firstWord.equals("deadline")) {
+            String[] arr = input.split("deadline ");
+            if (arr.length <= 1) {
+                throw new DukeException("The description of a deadline cannot be empty.");
+            } else {
+                String[] split = input.split("/by ");
+                if (split.length == 1) {
+                    throw new DukeException("By when??? You didn't include your deadline.");
+                } else {
+                    String tsk = split[0].split("deadline")[1].trim();
+                    String deadline = split[1].trim();
+                    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/M/yyyy");
+                    String dateString = deadline.split(" ")[0];
+                    String time = deadline.split(" ")[1];
+                    LocalDate date = LocalDate.parse(dateString, formatter);
+                    newTask = new Deadline(tsk, date, time);
+                }
+            }
+        } else {
+            String[] arr = input.split("event ");
+            if (arr.length == 1) {
+                throw new DukeException("The description of an event cannot be empty.");
+            } else {
+                String[] split = input.split("/at ");
+                if (split.length == 1) {
+                    throw new DukeException("At??? You didn't include the time of the event.");
+                } else {
+                    String at = split[1].trim();
+                    String tsk = split[0].split("event ")[1].trim();
+                    newTask = new Event(tsk, at);
+                }
+            }
+        }
+        return newTask;
+    }
+    public static void mainLogic(Scanner sc, ArrayList<Task> lst) {
+
+        while (true) {
```

Comment:
It might not be too good of practice to use "while (true)", as it is hard to tell what is your terminating condition (and there is a chance of an infinite loop)?